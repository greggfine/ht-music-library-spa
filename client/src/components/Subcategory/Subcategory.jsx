import React from "react";
import "./Subcategory.styles.scss";
import Types from "../Types/Types";
import useToggleState from "../../hooks/useToggleState";

const Subcategory = ({
  subcategoryName,
  categories,
  currentMainCategory,
  handleSetCurrentSubcategory
}) => {
  const [showMoreBtn, toggleShowMoreBtn] = useToggleState();
  const [showTypes, toggleShowTypes] = useToggleState();

  const handleSubcategoryClick = subcategoryName => {
    toggleShowMoreBtn();
    handleSetCurrentSubcategory(subcategoryName);
  };

  return (
    <>
      <div className="Subcategory">
        <li
          className="Subcategory-item"
          onClick={() => handleSubcategoryClick(subcategoryName)}
        >
          {subcategoryName}
        </li>
        {showMoreBtn && (
          <>
            <button onClick={() => toggleShowTypes()} className="show-more">
              Show more...
            </button>
            {showTypes && (
              <Types
                categories={categories}
                currentMainCategory={currentMainCategory}
                currentSubcategory={subcategoryName}
              />
            )}
          </>
        )}
      </div>
    </>
  );
};

export default Subcategory;
