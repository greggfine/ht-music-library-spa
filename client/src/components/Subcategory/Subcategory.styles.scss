.Subcategory {
  display: flex;
  flex-direction: column;
  .Subcategory-item {
    border: 1px solid #000;
    padding: 20px;
    margin-right: 10px;

    &:hover {
      cursor: pointer;
    }
  }

  .show-more {
    &:hover {
      cursor: pointer;
    }
  }
}

.track-list {
  //   display: flex;
}
