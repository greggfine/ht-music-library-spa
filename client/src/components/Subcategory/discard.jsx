import React, { useState } from "react";
import "./Subcategory.styles.scss";
import Types from "../Types/Types";
import useToggleState from "../../hooks/useToggleState";

const Subcategory = ({ subcategoryName, categories, currentMainCategory }) => {
  const [showMoreBtn, toggleShowMoreBtn] = useToggleState();
  const [showTypes, toggleShowTypes] = useToggleState();

  //   When we click on "Jazz", we toggle the showMoreBtn
  // We also LOAD up the first 25 tracks that have genre: 'jazz'

  return (
    <>
      <li className="Subcategory" onClick={toggleShowMoreBtn}>
        {subcategoryName}
      </li>
      {showMoreBtn && (
        <>
          <button onClick={() => toggleShowTypes()} className="show-more">
            Show more...
          </button>
          {showTypes && (
            <Types
              categories={categories}
              currentMainCategory={currentMainCategory}
              currentSubcategory={subcategoryName}
            />
          )}
        </>
      )}
    </>
  );
};

export default Subcategory;
