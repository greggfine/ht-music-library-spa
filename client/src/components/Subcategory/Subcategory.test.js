import React from 'react'
import {render, cleanup} from '@testing-library/react'
import Subcategory from './Subcategory'

afterEach(cleanup)

describe('<Subcategory />', () => {
    it("renders Subcategory component", () => {
        const {asFragment} = render(<Subcategory />)
        expect(asFragment()).toMatchSnapshot()
    })

    // it('inserts text in button', () => {
    //     const {getByTestId} = render(<Subcategory />)
    //     expect(getByTestId('showmorebtn')).toHaveTextContent('Show more...')
    // })

    /* 
        Are these Genres, etc. hard coded? or in Database?
        Where is this data?
        Do I need to make a new Collection?
            Or is this a subset of another collection?
    */
})