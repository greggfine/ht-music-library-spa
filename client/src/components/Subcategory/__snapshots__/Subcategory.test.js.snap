// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Subcategory /> renders Subcategory component 1`] = `
<DocumentFragment>
  <li
    class="Subcategory"
  />
</DocumentFragment>
`;
